sdk/src/main/scala/com/horizen/account/api/http/AccountTransactionApiRoute.scala

```diff
@@ -68,28 +73,144 @@ case class AccountTransactionApiRoute(override val settings: RESTApiSettings,
+                new SignatureData(msgSignature.getV, msgSignature.getR, msgSignature.getV)
+              )
+            )
+            validateAndSendTransaction(ethereumTransaction)
+          case None =>
+            ApiResponseUtil.toResponse(ErrorInsufficientBalance("ErrorInsufficientBalance", JOptional.empty()))
+        }
+      }
+    }
+  }
+
+  /**
+   * Create and sign a core transaction, specifying regular outputs and fee. Search for and spend proper amount of regular coins. Then validate and send the transaction.
+   * Return the new transaction as a hex string if format = false, otherwise its JSON representation.
+   */
+  def createEIP1559Transaction: Route = (post & path("createEIP1559Transaction")) {
+    entity(as[ReqCreateEIP1559Transaction]) { body =>
+      val ethTx = new EthereumTransaction(
+        body.payload.chainId,
+        body.payload.to.orNull,
+        body.payload.nonce,
+        body.payload.gasLimit,
+        body.payload.maxPriorityFeePerGas,
+        body.payload.maxFeePerGas,
+        body.payload.value,
+        body.payload.data,
+        if (body.payload.signature_v.isDefined)
+          new SignatureData(
+            body.payload.signature_v.get,
+            body.payload.signature_r.get,
+            body.payload.signature_s.get)
+        else
+          null
+      )
+      // lock the view and try to create CoreTransaction
+      applyOnNodeView { sidechainNodeView =>
+        validateAndSendTransaction(ethTx)
+      }
+    }
+  }
+
+  /**
+   * Create a legacy evm transaction, specifying inputs.
+   */
+  def createLegacyTransaction: Route = (post & path("createLegacyTransaction")) {
+    entity(as[ReqCreateLegacyTransaction]) { body =>
+      //  TODO uncomment once EVM-161 is merged
+      val ethTx = new EthereumTransaction(
+        body.payload.to.orNull,
+        body.payload.nonce,
+        body.payload.gasPrice,
+        body.payload.gasLimit,
+        body.payload.value,
+        body.payload.data,
+        if (body.payload.signature_v.isDefined)
+          new SignatureData(
+            body.payload.signature_v.get,
+            body.payload.signature_r.get,
+            body.payload.signature_s.get)
+        else
+          null
+      )
+      // lock the view and try to send the tx
+      applyOnNodeView { sidechainNodeView =>
+        validateAndSendTransaction(ethTx)
+      }
+    }
+  }
+
+  /**
+   * Create a raw evm transaction, specifying the bytes.
+   */
+  def sendRawTransaction: Route = (post & path("createRawTransaction")) {
+    entity(as[ReqCreateRawTransaction]) { body =>
+      val ethTx = EthereumTransactionSerializer.getSerializer.parseBytes(body.payload)
+      // lock the view and try to create CoreTransaction
+      applyOnNodeView { sidechainNodeView =>
+        validateAndSendTransaction(ethTx)
+      }
+    }
+  }
+
+  def signTransaction: Route = (post & path("signTransaction")) {
+    entity(as[ReqCreateRawTransaction]) {
```

Comment:
I expect that we need to add one more argument - account address.

So, we need to sign Tx with the real account key that belongs to the node.
Signing with random key has no sence, because tx validation will fail because of balance is 0

Reply:
Would it make sense in your opinion to generally add an optional param `from` which can be used to create the signature, so that it goes like this

`is from specified? 
   yes => look for the corresponding account and use that 
   no  => pick the first account you find which satisfies needed conditions (gas cost in the future, and value of tx)`

Reply:
If user want to choose first one from the wallet there is a endpoint for it `sendCoinsToAddress`

Reply:
I was thinking of unifying the behavior between those three functions instead, so that the user knows that it's either the first one that fits or the specific one they chose, no matter which function they're calling to do it

Reply:
In the new changes, the behavior is the same for all the routes with ethereum transactions. This makes it easy for the user to handle these, since they all act in the same way